### PR TITLE
feat(datasource): rename dotnet to `dotnet-version`

### LINKED ISSUE #20233: Rename `dotnet` datasource to `dotnet-version`
### What would you like Renovate to be able to do?

We should do this to be in line with our other datasources and don't confuse users with `nuget` datasource

### If you have any ideas on how this should be implemented, please tell us here.

just rename and migrate

### Is this a feature you are interested in implementing yourself?

Maybe

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- rename and migrate datasource `dotnet` to `dotnet-version`

## Context
- Closes: #20233 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/migrations/custom/datasource-migration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { DatasourceMigration } from './datasource-migration';
```
Comment (HonkingGoose) [inline lib/config/migrations/custom/datasource-migration.spec.ts:1]: PR titles should use the present tense, like this:

```diff
- refactor(datasource): renamed dotnet to `dotnet-version`
+ refactor(datasource): rename dotnet to `dotnet-version`
```

Comment (RahulGautamSingh) [reply lib/config/migrations/custom/datasource-migration.spec.ts:1]: Typo 😛

File: lib/modules/datasource/dotnet-version/index.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
 } from './schema';
 
 export class DotnetDatasource extends Datasource {
```
Comment (viceice) [inline lib/modules/datasource/dotnet-version/index.ts:14]:
```suggestion
 export class DotnetVersionDatasource extends Datasource {
```

**Review by viceice — COMMENTED** [review]
just a small rename, even it causes a lot more code changes.

**Review by JamieMagee — DISMISSED** [review]
Sorry about this 🤦

**Review by JamieMagee — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by bourquep** [comment]
Since this Renovate release (running Renovate `34.132.0` to be precise), I've been getting these errors for all my dotnet repos:

> There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop MRs until it is resolved.
> Location: `renovate.json`
> Error type: The renovate configuration file contains some invalid settings
> Message: `Invalid configuration option: dotnet-version`

My `renovate.json`:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "labels": [
    "dependencies"
  ],
  "reviewers": [
    "@redacted"
  ],
  "gitAuthor": "Renovate Bot <redacted>",
  "ignorePaths": [
    ".devcontainer/docker-compose.yml"
  ],
  "dotnet": {
    "packageRules": [
      {
        "matchDatasources": [
          "nuget"
        ],
        "registryUrls": [
          "https://api.nuget.org/v3/index.json",
          "https://gitlab.com/api/v4/groups/redacted/-/packages/nuget/index.json"
        ]
      },
      {
        "matchPackagePrefixes": [
          "Studyo"
        ],
        "groupName": "studyo"
      },
      {
        "matchPackagePrefixes": [
          "Microsoft"
        ],
        "groupName": "microsoft"
      },
      {
        "matchPackagePrefixes": [
          "Hangfire"
        ],
        "groupName": "hangfire"
      },
      {
        "matchPackagePrefixes": [
          "Serilog"
        ],
        "groupName": "serilog"
      },
      {
        "matchPackagePrefixes": [
          "prometheus-net"
        ],
        "groupName": "prometheus"
      },
      {
        "matchPackagePatterns": [
          "Test",
          "test",
          "xunit",
          "Moq",
          "coverlet",
          "codeclimate",
          "jetbrains",
          "dotnet-reportgenerator"
        ],
        "groupName": "tests"
      }
    ]
  },
  "docker": {
    "versioning": "docker",
    "packageRules": [
      {
        "matchPackagePrefixes": [
          "mcr.microsoft.com/dotnet"
        ],
        "groupName": "dotnet-image"
      }
    ]
  }
}
```

**Comment by RahulGautamSingh** [comment]
I renamed `dotnet` property in globalConfig to `dotnet-version` with these changes but I will have to add a PR to revert it as the `dotnet` is also a language and those configs are also getting migrated.
Sorry for the trouble and thanks for the comment.

**Pull request merged**